java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -0,0 +1,206 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.bazel.tests.integration;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+
+
+public class WorkspaceDriver {
+
+  private static File tmp;
+  private static Map<String, File> bazelVersions;
+  private static File runfileDirectory = new File(System.getenv("TEST_SRCDIR"));
+
+  private File currentBazel = null;
+
+  /**
+   * The current workspace.
+   */
+  protected File workspace = null;
```

Comment:
should this be protected?
this sounds like we need this to be private (and understand if the test that currently needs it can do without), be public (don't like public mutable variables) or expose a relevant API